src/adapter/src/coord/sequencer.rs

```diff
@@ -471,6 +472,32 @@ impl Coordinator {
         }
     }
 
+    // Utilities used by `sequence` functions
```

Comment:
This seems a little weird. We've got a bunch of utility funcs at the end of the `impl` block, if you instead wanted to toss this down there?